org-formation/090-systems-manager/Scheduled-Script-Automation.yaml

```diff
@@ -0,0 +1,297 @@
+                      'TargetTagName' : [TargetTagName],
+                      'TargetTagValue' : [TargetTagValue],
+                      'ScriptUrl' : [ScriptUrl],
+                      'ScriptParameters' : [ScriptParameters]
+                  },
+                  TargetLocations=[
+                      {
+                          'Accounts': TargetAccountsArray,
+                          'Regions': TargetRegionIdsArray,
+                          'TargetLocationMaxConcurrency': f'{TargetLocationMaxConcurrency}',
+                          'TargetLocationMaxErrors': f'{TargetLocationMaxErrors}',
+                          'ExecutionRoleName': f'{ExecutionRoleName}'
+                      }
+                  ]
+              )
+              print(response)
+      Environment:
+        Variables:
+            # Fn::EnumTargetAccounts is specific to org-formation. It returns all ScriptExecutionMemberBinding account IDs defined in _tasks.yaml
+            TargetAccounts:
+              Fn::Join:
+                - ","
+                - - Fn::EnumTargetAccounts ScriptExecutionMemberBinding ${account}
+            TargetRegionIds: !Ref TargetRegionIds
+            TargetLocationMaxConcurrency: !Ref TargetLocationMaxConcurrency
+            TargetLocationMaxErrors: !Ref TargetLocationMaxErrors
+            ExecutionRoleName: !Ref ExecutionRoleName
+            MasterAccountID: !Sub ${AWS::AccountId}
+            AutomationDocumentScriptExecution: !Ref AutomationDocumentScriptExecution
+            ScriptUrl: !Ref ScriptUrl
+            ScriptParameters: !Ref ScriptParameters
+            TargetTagName: !Ref TargetTagName
+            TargetTagValue: !Ref TargetTagValue
+      FunctionName: MultiAccountScriptExecution
+      Handler: index.handler
+      Role: !GetAtt AWSLambdaSSMMultiAccountScriptExecutionRole.Arn
+      Runtime: python3.12
+  AutomationAdministrationServiceRole:
+    Type: AWS::IAM::Role
+    Condition: CreateAutomationAdministrationRoleCondition
+    Properties:
+      RoleName: AWS-SystemsManager-AutomationAdministrationRole
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service: ssm.amazonaws.com
+          Action:
+          - sts:AssumeRole
+      Path: "/"
+      Policies:
+      - PolicyName: AssumeRole-AWSSystemsManagerAutomationExecutionRole
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+          - Effect: Allow
+            Action:
+            - sts:AssumeRole
+            Resource:
+              Fn::Sub: arn:${AWS::Partition}:iam::*:role/AWS-SystemsManager-AutomationExecutionRole
+          - Effect: Allow
+            Action:
+            - organizations:ListAccountsForParent
+            Resource:
+            - "*"
+  AutomationDocumentScriptExecution:
+    Type: "AWS::SSM::Document"
+    Properties:
+      DocumentType: "Command"
```

Comment:
It looks like you copy and pasted most of this from  the `Automation` DocumentType in `Scheduled-Patch-Automation.yaml` file?  I don't think the schema for that is the same as the `Command` DocumentType.  I think you will need to update to match the schema for the `Command` DocumentType https://docs.aws.amazon.com/systems-manager/latest/userguide/documents-command-ssm-plugin-reference.html#shared-inputs

Reply:
According to [the doc's](https://docs.aws.amazon.com/systems-manager/latest/userguide/documents-schemas-features.html) the schemaVersion 2.2 for Documents has the fields, `description`, `parameters`, and `mainSteps`, which is what we are using here.  (Your code uses schemaVersion 0.3, which is [required](https://docs.aws.amazon.com/systems-manager/latest/userguide/documents-schemas-features.html) for Documents of type Automation.)

If I am missing something (if you see something wrong in this code) please say what the mistake is.

Reply:
I think this document uses the correct plug-ins, i.e., this document is a `Command` document and uses the `aws:runShellScript` plugin, which is listed as a valid plug-in for the `Command` document type in the document you linked.  So I don't see any problem there.   However I see another potential problem:  [This](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-automation.html) doc' says that Automation uses Automation runbooks and, "Runbooks are Systems Manager documents of type Automation, as opposed to Command, Policy, Session documents. "  So I think I have to switch from a `Command` document  type to an `Automation` document type, which would involve changing the document parameters.

Reply:
@zaro0508 The PR is now updated to use document type `Automation`, the compatible plug-in type, `aws:runCommand`, and the Document, `AWS-RunShellScript`.  I tested the CF template in a dev account to make sure the parameters are right.